py-tests/pyspark_tests.py

```diff
@@ -0,0 +1,642 @@
+        try:
+            pinged = ping_with_timeout_1_second(client)
+        except:
+            pass
+        i = i + 1
+        time.sleep(1)
+    return pinged
+
+def wait_until_write_read_kv(client, times=20):
+
+    written = False
+    i = 0
+    while not written and i < times:
+        try:
+            store_obj_timeout_1_second(client, 'temp_bucket', 'temp_key', 'temp_data')
+            written = True
+        except:
+            pass
+        i = i + 1
+        time.sleep(1)
+    read = False
+    i = 0
+    while not read and i < times:
+        try:
+            temp_data = read_obj_timeout_1_second(client, 'temp_bucket', 'temp_key')
+            if temp_data == 'temp_data':
+                read = True
+        except:
+            pass
+        i = i + 1
+        time.sleep(1)
+    if written == True and read == True:
+        return True
+    else:
+        return False
+
+def ts_write_verify(client, table_name, query, expected, times = 10):
+
+    read = False
+    i = 0
+    while not read and i < times:
+        try:
+            temp_read = client.ts_query(table_name, query)
+            #print(temp_read.rows)
+            #print(expected)
+            if temp_read.rows == expected:
+                read = True
+        except:
+            pass
+        i = i + 1
+        time.sleep(1)
+    if read:
+        return True
+    else:
+        return False
+
+def df_write_retry(df, table_name, times = 20):
+
+    written = False
+    i = 0 
+    while not written and i < times:
+        try:
+            df.write.format('org.apache.spark.sql.riak').mode('Append').save(table_name)
+            written = True
+        except:
+            pass
+        i = i + 1
+        time.sleep(1)
+    if written:
+        return True
+    else:
+        return False
+
+def df_read_retry(sql_context, table_name, temp_filter, useLong = False, useRange = False, times = 20):
+
+    read = False
+    i = 0 
+    while not read and i < times:
+        try:
+            if useLong:
+                if useRnage:
+                    read_df = sql_context.read.format("org.apache.spark.sql.riak").option("spark.riakts.bindings.timestamp", "useLong").option("spark.riak.partitioning.ts-range-field-name", "datetime").load(table_name).filter(temp_filter)
+                    read = True
+                else:
+                    read_df = sql_context.read.format("org.apache.spark.sql.riak").option("spark.riakts.bindings.timestamp", "useLong").load(table_name).filter(temp_filter)
+                    read = True
+
+
+
+            else:
+                if useRange:
+                    read_df = sql_context.read.format("org.apache.spark.sql.riak").option("spark.riak.partitioning.ts-range-field-name", "datetime").load(table_name).filter(temp_filter)
+                    read = True
+                else:
+                    read_df = sql_context.read.format("org.apache.spark.sql.riak").load(table_name).filter(temp_filter)
```

Comment:
If I understand the useX options and their sql_context counterparts, the setting of options and the execution of the query seem to be better split into two distinct set of steps. Otherwise, there is nested if's and duplicate code such as all branches having the method chaining ending w/ `.load(table_name).filter(temp_filter)` .